go/tools/builders/checker_main.go

```diff
@@ -44,39 +44,93 @@ import (
 // do not return an error so as not to unnecessarily interrupt builds.
 func run(args []string) error {
 	archiveFiles := multiFlag{}
+	importMapIn := multiFlag{}
+	srcs := multiFlag{}
 	flags := flag.NewFlagSet("checker", flag.ContinueOnError)
 	flags.Var(&archiveFiles, "archivefile", "Archive file of a direct dependency")
+	flags.Var(&importMapIn, "importmap", "Import maps of a direct dependency")
+	flags.Var(&srcs, "src", "A source file being compiled")
+	vetTool := flags.String("vet_tool", "", "The vet tool")
+	stdLib := flags.String("stdlib", "", "The directory containing stdlib packages")
+	packageList := flags.String("package_list", "", "The file containing the list of standard library packages")
 	if err := flags.Parse(args); err != nil {
 		log.Println(err)
 		return nil
 	}
-	goroot, ok := os.LookupEnv("GOROOT")
-	if !ok {
-		log.Fatalf("GOROOT not set")
+
+	packageFile, importMap, err := parseArchivesAndImportMap(archiveFiles, importMapIn)
+	if err != nil {
+		log.Printf("error parsing arguments: %v", err)
+		return nil
+	}
+	if enableVet {
```

Comment:
Continuing the thread about wrapping vet as an `Analysis`, I asked Alan about this, and he wants to keep findings they way they are. It's important to be able to programmatically consume findings alongside the other data produced by the API.

Vet as it stands doesn't fit that model (though we plan to do some refactoring so it will). I expect in the future we'll want to wrap external tools in general, so we may want another abstraction, but no change needed for now.

Reply:
Ack.